pkg/dependency/parser/java/pom/artifact.go

```diff
@@ -42,7 +49,13 @@ func newArtifact(groupID, artifactID, version string, licenses []string, props m
 }
 
 func (a artifact) IsEmpty() bool {
-	return a.GroupID == "" || a.ArtifactID == "" || a.Version.String() == ""
+	if a.GroupID == "" || a.ArtifactID == "" {
+		return true
+	}
+	if a.Version.String() == "" {
+		emptyVersionWarn()
```

Comment:
We may want to show warnings with group id and artifact id. Can it be overwhelming?

Reply:
I decided to show the warning once to avoid spam in logs
do you think we should still show all dependencies without the version?

Reply:
One warning without dependency details and other debug logs with group id and artifact id may be a good compromise.

Reply:
updated in https://github.com/aquasecurity/trivy/pull/7520/commits/9424f139dea73a39b22012de613dff9085efc8f5